File: src/01-easy/mime-type/solution.kt
fun main() {
    val n = readLine()!!.trim().toInt()
    val q = readLine()!!.trim().toInt()
    val m = mutableMapOf<String,String>()
    repeat(n) {
        val (ext, mt) = readLine()!!.split(" ")
        m[ext.lowercase()] = mt
    }
    repeat(q) {
        val f = readLine()!!
        val dot = f.lastIndexOf('.')
        if (dot == -1) println("UNKNOWN")
        else println(m[f.substring(dot + 1).lowercase()] ?: "UNKNOWN")
    }
}

File: src/01-easy/mime-type/solution.fs
open System
open System.Collections.Generic
let n = Console.ReadLine() |> int
let q = Console.ReadLine() |> int
let m = Dictionary<string,string>()
for _ in 1..n do
    let p = Console.ReadLine().Split(' ')
    m.[p.[0].ToLower()] <- p.[1]
for _ in 1..q do
    let f = Console.ReadLine()
    let dot = f.LastIndexOf('.')
    if dot = -1 then printfn "UNKNOWN"
    else
        let ext = f.Substring(dot + 1).ToLower()
        match m.TryGetValue(ext) with
        | true, v -> printfn "%s" v
        | _ -> printfn "UNKNOWN"
